iterator_practice.cpp

```diff
@@ -1,16 +1,13 @@
 #include <iostream>
-#include <iterator> // for iterator
 #include <vector> // for vectors 
 
-using namespace std;
-
-int main()
-{
-    vector<int> ar = {2,4,6,8,10};
-    vector<int> ::iterator ptr;
-    cout << "My vector elements are: ";
-    for (ptr = ar.begin(); ptr < ar.end (); ptr++)
-        cout << *ptr<< " ";
-
-    return 0; 
-}
\ No newline at end of file
+int main() {
+  const std::vector<int> evens{2,4,6,8,10};
```

Comment:
I'm suggesting a couple of changes here:

1. You never change this variable after making it, so I mark it const. This tells the compiler to make sure you never change it. If you try, you'll get an error. General rule: if you don't want to change something, make it const, so the compiler ensures you really don't change it

2. We prefer descriptive names, so you can tell what a variable contains. So I suggest calling the variable `evens`.

3. I suggest using the modern c++ way of creating an object from its constructor (the constructor is a function that initialized an object). Instead of =, put the initial arguments in curly braces.

But actually, I should have done `std::array<int, 5> evens{{2, 4, 6, 8, 10}};` here (and change the include to `#include <array>` above). When you know how many elements you have, and that number doesn't change, use `std::array` instead of `std::vector`. Because you can resize a `std::vector`, it's slower to use, and the compiler won't throw an error if you give it the wrong number of elements. 

Also, a detail: I maybe should have used double curly braces. I wanted to initialize an array with an array, but instead initialized it with 5 ints instead. Not sure if it matters...gives the same result here.